charts/opentelemetry-demo-datadogv2/exporter.yaml

```diff
@@ -0,0 +1,66 @@
+ingest: false
+opentelemetry-demo:
+  opentelemetry-collector:
+    mode: "daemonset"
+    config:
+      receivers:
+        otlp:
+          protocols:
+            http:
+            grpc:
+        # The hostmetrics receiver is required to get correct infrastructure metrics in Datadog.
+        hostmetrics:
+          collection_interval: 10s
+          scrapers:
+            paging:
+              metrics:
+                system.paging.utilization:
+                  enabled: true
+            cpu:
+              metrics:
+                system.cpu.utilization:
+                  enabled: true
+            disk:
+            filesystem:
+              metrics:
+                system.filesystem.utilization:
+                  enabled: true
+            load:
+            memory:
+            network:
+            processes:
+        # The prometheus receiver scrapes metrics needed for the OpenTelemetry Collector Dashboard.
+        prometheus:
+          config:
+            scrape_configs:
+              - job_name: 'otelcol'
+                scrape_interval: 10s
+                static_configs:
+                  - targets: [ '0.0.0.0:8888' ]
+
+      processors:
+        batch:
+          # Datadog APM Intake limit is 3.2MB. Let's make sure the batches do not
+          # go over that.
+          send_batch_max_size: 1000
+          send_batch_size: 100
+          timeout: 10s
+
+      exporters:
+        datadog:
+          api:
+            site: datadoghq.com
+            key: ${DD_API_KEY}
```

Comment:
This would set `DD_API_KEY` in plain text format in the config. It is generally recommended to have a secret for api keys. 
See https://github.com/DataDog/opentelemetry-helm-charts/blob/e7404a0c87dae9f5b6fc8e2cd84f29ed9f25d19c/charts/opentelemetry-demo-datadog/templates/otelcol_operator.yaml#L15 


https://github.com/DataDog/opentelemetry-helm-charts/blob/e7404a0c87dae9f5b6fc8e2cd84f29ed9f25d19c/charts/opentelemetry-demo-datadog/templates/secret.yaml

Reply:
This would be good to have . We can avoid in this repo it is complicated. When deploying to staging we use vault

Reply:
Hmm... Our [docs](https://docs.datadoghq.com/tracing/trace_collection/open_standards/otel_collector_datadog_exporter/#using-kubernetes) reference an [old file](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/2c32722e37f171bab247684e7c07e824429a8121/exporter/datadogexporter/examples/k8s-chart/configmap.yaml). [Even main seems to have this in plaintext 👀 ](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/datadogexporter/examples/k8s-chart/configmap.yaml).

This seems like a problem?

Reply:
Are we not just meant to `--set` this on installation of the chart? The subcharts themselves handle it properly? Otherwise the subcharts themselves are broken, no?